pkg/providers/vsphere/session/session_vm_update.go

```diff
@@ -584,7 +585,14 @@ func (s *Session) prePowerOnVMReconfigure(
 
 	defaultConfigSpec := &vimtypes.VirtualMachineConfigSpec{}
 	if !apiEquality.Semantic.DeepEqual(configSpec, defaultConfigSpec) {
-		vmCtx.Logger.Info("Pre PowerOn Reconfigure", "configSpec", configSpec)
+		var w bytes.Buffer
+		enc := vimtypes.NewJSONEncoder(&w)
+		if err := enc.Encode(configSpec); err != nil {
+			vmCtx.Logger.Error(err, "Failed to marshal ConfigSpec to JSON")
+			vmCtx.Logger.Info("Pre PowerOn Reconfigure", "configSpec", configSpec)
+		} else {
+			vmCtx.Logger.Info("Pre PowerOn Reconfigure", "configSpec", w.String())
```

Comment:
@sreyasn / @bryanv This will be a helpful change as it will include the actual type information from the ConfigSpec as part of the logged reconfigure. We should do something similar for resize.